lib/core/io.js

```diff
@@ -225,8 +224,8 @@ class FetchReader extends Reader {
 	async init() {
 		super.init();
 		if (isHttpFamily(this.url) && !this.preventHeadRequest) {
-			const response = await sendFetchRequest(HTTP_METHOD_HEAD, this.url, this.options);
-			this.size = Number(response.headers.get(HTTP_HEADER_CONTENT_LENGTH));
+			const response = await sendFetchRequest(HTTP_METHOD_GET, this.url, this.options, { range: "bytes=0-0"});
+			this.size = Number(response.headers.get(HTTP_HEADER_CONTENT_RANGE).split('/')[1]);
 			if (!this.forceRangeRequests && this.useRangeHeader && response.headers.get(HTTP_HEADER_ACCEPT_RANGES) != HTTP_RANGE_UNIT) {
```

Comment:
This test should be run before the implementation you suggest.

Reply:
Done.